server/rogue/auth-strategies/client-credentials.js

```diff
@@ -0,0 +1,197 @@
+   */
+  constructor(opts = {}) {
+    super();
+    // Amount of milliseconds to wait before checking if we need to renew the token
+    this.tickerInterval = opts.tickerInterval || config.defaults.tickerInterval;
+    /**
+     * If we can't renew the token in maxReconnectAttempts retries, wait this many milliseconds
+     * before trying to reconnect again
+     */
+    this.reconnectWaitPeriod = opts.reconnectWaitPeriod || config.defaults.reconnectWaitPeriod;
+    /**
+     * Should the token be preemptively renewed?
+     * In the case of Northstar we Should, because it sends 'expires_in' instead of 'expires_at'
+     * generating a possible race condition. Described below.
+     * @see https://www.npmjs.com/package/simple-oauth2#access-token-object
+     */
+    this.autoRenewToken = opts.autoRenewToken || config.defaults.autoRenewToken;
+    // How many seconds before the token expires we should attempt to renew?
+    this.renewWindow = opts.renewWindow || config.defaults.renewWindow;
+    // Maximum amount of linear retries to attemp before a waiting period
+    this.maxReconnectAttempts = opts.maxReconnectAttempts || 1;
+    // It increases on each attempt to reconnect
+    this.reconnectAttempts = 0;
+    /**
+     * We might need in order to ask for specific scopes
+     * @see https://github.com/lelylan/simple-oauth2/tree/013ddf2a3876c57c7b706599d8688f4bf48ea660#client-credentials-flow
+     */
+    this.tokenConfig = opts.tokenConfig || {};
+    this.oauthClient = oauth2.create(config.credentials);
+    // Used for testing
+    this.setInterval = opts.setInterval || setInterval;
+  }
+  /**
+   * setup - Initializes the setup of the Oauth token and sets listeners to respond
+   * to 'token-set' and 'token-error' events.
+   *
+   * @return {Promise}
+   */
+  setup() {
+    // token-set listener
+    this.on('token-set', () => {
+      logger.info('ClientCredentials.setup: token has been set');
+      // TODO: Maybe create a separate token refresh class that encapsulates renewing tokens.
+      if (this.autoRenewToken && !this.ticker) {
+        this.setTicker(this.renewExpiredToken.bind(this));
+      }
+      // Some cleanup, TODO: maybe move to cleanup method?
+      this.reconnectAttempts = 0;
+    });
+    // token-error listener
+    this.on('token-error', (error) => {
+      logger.error(`ClientCredentials.setup: Access Token Error. Error Message: ${error.message}`);
+      this.reconnect();
+    });
+
+    // Get and set new token
+    return this.renewToken();
+  }
+  /**
+   * getAuthorizedClient - Sets the Authorization header and Bearer token in the superagent client.
+   *                       All the requests sent through this client will use this header.
+   *
+   * @return {Object}  authenticated superagent client
+   */
+  getAuthorizedClient() {
+    return superagent.use((req) => {
+      req.set('Authorization', `Bearer ${this.token.access_token}`);
+      req.accept('application/json');
```

Comment:
Nitpicky: this doesn't seem like something that `ClientCredentials` should enforce. You could use an OAuth token to ask for any sort of resource, not just JSON.